wp-multi-network/includes/classes/class-wp-ms-networks-api.php

```diff
@@ -0,0 +1,859 @@
+
+		$fields_update = $this->update_additional_fields_for_object( $network, $request );
+
+		if ( is_wp_error( $fields_update ) ) {
+			return $fields_update;
+		}
+
+		$request->set_param( 'context', 'edit' );
+
+		$response = $this->prepare_item_for_response( $network, $request );
+
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Checks if a given request has access to delete a network.
+	 *
+	 * @since 2.2.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|bool True if the request has access to delete the item, error object otherwise.
+	 */
+	public function delete_item_permissions_check( $request ) {
+		if ( 0 === get_current_user_id() ) {
+			return false;
+		}
+
+		$network = $this->get_network( $request['id'] );
+		if ( is_wp_error( $network ) ) {
+			return $network;
+		}
+
+		return current_user_can( 'delete_network', $network->id );
+	}
+
+	/**
+	 * Deletes a network.
+	 *
+	 * @since 2.2.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response Response object on success, or error object on failure.
+	 */
+	public function delete_item( $request ) {
+		$network = $this->get_network( $request['id'] );
+		if ( is_wp_error( $network ) ) {
+			return $network;
+		}
+
+		$force = isset( $request['force'] ) ? (bool) $request['force'] : false;
+
+		$request->set_param( 'context', 'edit' );
+
+		$previous = $this->prepare_item_for_response( $network, $request );
+		$result = delete_network( $network->id, $force );
+		$response = new WP_REST_Response();
+		$response->set_data(
+			array(
+				'deleted' => true,
+				'previous' => $previous->get_data(),
+			)
+		);
+
+		if( is_wp_error( $result ) ){
+			return $result;
+		} elseif ( ! $result ) {
+			return new WP_Error(
+				'rest_cannot_delete', __( 'The network cannot be deleted.', 'wp-multi-network' ), array(
+					'status' => 500,
+				)
+			);
+		}
+
+		/**
+		 * Fires after a network is deleted via the REST API.
+		 *
+		 * @since 2.2.0
+		 *
+		 * @param WP_Network       $network  The deleted network data.
+		 * @param WP_REST_Response $response The response returned from the API.
+		 * @param WP_REST_Request  $request  The request sent to the API.
+		 */
+		do_action( 'rest_delete_network', $network, $response, $request );
+
+		return $response;
+	}
+
+	/**
+	 * Prepares a single network output for response.
+	 *
+	 * @since 2.2.0
+	 *
+	 * @param WP_Network      $network Network object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $network, $request ) {
+		$data = array(
+			'id'                 => (int) $network->id,
+			'path'               => $network->path,
+			'domain'             => $network->domain,
+			'cookie_domain'      => $network->cookie_domain,
+			'site_name'          => $network->site_name,
+			'blog_id'            => (int) $network->blog_id,
+		);
```

Comment:
The network data should be added to the array one by one, based on whether it is present in the schema properties. Also `blog_id` should rather be called `site_id` now, as it's the current name and can also be accessed like that from the object.